### PR TITLE
Xlassix/dao

### PR DESCRIPTION
**Pull request by xlassix**

### CODE REVIEW

File: contracts/DepositManager.sol [hunk 1]
```diff
@@ -95,15 +95,15 @@ contract DepositManager {
...
      */
-    function transferAmount(
+    function approveSpender(
         address _coinAddress,
         address wallet,
         uint256 amount
     ) external onlyOwner {
         require(amount <= IERC20(_coinAddress).balanceOf(address(this)));
-        IERC20(_coinAddress).transfer(wallet, amount);
+        IERC20(_coinAddress).approve(wallet, amount);
```
Comment (nandit123) [inline contracts/DepositManager.sol:106]: where is the function for transfer?

**Review by nandit123 — COMMENTED** [review]
I have put up some comments, overall structure looks good

File: contracts/DAO/LighthouseGovernance.sol [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        Governor("LighthouseGovernor")
+        GovernorSettings(
+            _votingDelay,
+            _votingPeriod, 0)
+        GovernorVotes(_token)
+        GovernorTimelockControl(_timelock)
+    {}
+
+    function quorum(uint256 blockNumber) public pure override returns (uint256) {
+        return 40e18;
```
[the remark this answers is not in the document]
Comment (xlassix) [reply contracts/DAO/LighthouseGovernance.sol:23]: its a standard, there are usually who approach the first using percentage and the other a quote of the tokens most vote 
in this case, it's fixed at 40% of the max supply(a fixed value).

File: contracts/DepositManager.sol (hunk 1 above)
Comment (xlassix) [reply contracts/DepositManager.sol:106]: there are two approaches when working with funds on a smart contract.
1. the push approach sends the funds directly
2. the pull approach gives access to a party to spend an amount of token.

while auditing and working on AAVE, the pull approach seems to be more advantageous for a bunch of reasons and in addition we could validate where the funds are coming from as the receiver in a contract. basically, after given the approval to spend any amount, the spender is allowed to send/spend that amount on behalf of who gave the approval

Comment (nandit123) [reply contracts/DepositManager.sol:106]: So will the AAVE protocol execute the transaction by itself in this case?

File: contracts/DAO/LighthouseGovernance.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (nandit123) [reply contracts/DAO/LighthouseGovernance.sol:23]: So will this value be fixed? or can we change it later ?

File: configHardhat.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    blockConfirmations?: number
+  }
+  
+  export interface networkConfigInfo {
+    [key: string]: networkConfigItem
+  }
+
+  export const developmentChains = ["hardhat", "localhost"]
+  
+  // Governor Values
```
[the remark this answers is not in the document]
Comment (nandit123) [reply configHardhat.ts:12]: @xlassix

[the remark this answers is not in the document]
Comment (nandit123) [reply configHardhat.ts:12]: Implement the function to update these values

**Pull request closed without merging**